pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1113,6 +1113,16 @@ func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.Clu
 		return reconcile.Result{}, err
 	}
 
+	// We do not yet support deprovision for BareMetal, for now skip deprovision and remove finalizer.
+	if cd.Spec.Platform.BareMetal != nil {
+		cdLog.Info("skipping deprovision for BareMetal cluster, removing finalizer")
+		err = r.removeClusterDeploymentFinalizer(cd)
```

Comment:
Oh, one more thing. We should limit the scope of variables as much as possible. Use `err := `.